lib/msf/base/simple/auxiliary.rb

```diff
@@ -132,12 +160,23 @@ def check_simple(opts)
   #
   # Job run proc, sets up the module and kicks it off.
   #
-  def self.job_run_proc(ctx)
+  def self.job_run_proc(ctx, &block)
     mod = ctx[0]
+    run_uuid = ctx[1]
     begin
       mod.setup
       mod.framework.events.on_module_run(mod)
-      mod.run
+      begin
+        mod.framework.running << run_uuid
+        mod.framework.ready.delete run_uuid
+        result = block.call(mod)
+        mod.framework.results[run_uuid] = {result: result}
+      rescue Exception => e
+        mod.framework.results[run_uuid] = {error: e.to_s}
+        raise
```

Comment:
Should this raise a specific type or pass `e`?

Reply:
`$!` is the last exception rescued and is what `raise` raises when no arguments are passed inside a `rescue` clause.